src/pages/settings/Security/TwoFactorAuth/Steps/VerifyStep.js

```diff
@@ -45,8 +46,12 @@ function VerifyStep({account, session}) {
         if (!account.requiresTwoFactorAuth) {
             return;
         }
-        setStep(CONST.TWO_FACTOR_AUTH_STEPS.SUCCESS);
-    }, [account.requiresTwoFactorAuth, setStep]);
+        navigation.dispatch({
+            ...CommonActions.goBack(),
+            source: navigation.getState().key,
+        });
```

Comment:
I can't find any more usage of `navigation.dispatch` across the app, why it's needed here?

Reply:
The reason is similar to https://github.com/Expensify/App/pull/28895#discussion_r1356663104, but instead of needing just one route in history to be removed, it needs two, `codes` and `verify`.

So if the user goes `settings` -> `codes` -> `verify` -> `success`, and clicks back button it will get him to `settings` page.